### ISSUE TITLE
Cannot Add Custom Search Engine in Firefox Focus

### ISSUE THREAD

**Issue by st3fan** [mozilla-mobile/focus-android#4415]
> Copied from https://bugzilla.mozilla.org/show_bug.cgi?id=1590327

Steps to reproduce:

Add Custom Search Engine in Firefox Focus. I tried to add MetaGER with string "https://metager.de/meta/meta.ger3?focus=web&encoding=utf8&lang=all&eingabe=%s".

Actual results:

Is does not appear in the list of search engines. If I try to add the same again, Firefox tells me that there is an existing search engine with this name.

Expected results:

Should appear in the list of search engines and should be selectable.

**Comment by woodruffandtheschnibbleofazimuth** [comment]
I have the same issue also on the last version of Firefox Focus.

**Comment by goforbg** [comment]
Same issue still persists, turns out it turns up only after an app restart.
The language code is beautifully written to update when Shared Preferences are changed, but I think the CustomSearchEngineProvider.kt class runs only once during startup and that's why the list isn't updated. I'm trying to solve this, will post an update here soon.

**Comment by srustamo** [comment]
> Same issue still persists, turns out it turns up only after an app restart.

I don't see the newly added search engine even after a restart. v 8.11.1

**Comment by tkbremnes** [comment]
The search engine appears on _device_ restart on my phone.

**Comment by davhel** [comment]
Duplicate of #4394 
I can also reproduce the issue on android 7.1.1. using v8.12.0
Removal of queries is affected as well 

@zhuperman opened a pull request to fix this issue but it never got merged
https://github.com/mozilla-mobile/focus-android/pull/4487

**Comment by lobontiumira** [comment]
This is no longer reproducible on the latest Focus builds.
I'll close this ticket.

**Comment by ykiran** [comment]
Still having the issue. 
Not able to have the Brave search engine.
![image](https://user-images.githubusercontent.com/5526410/158015517-55e7e074-9cde-4a3f-83e1-d1c6be44bb69.jpeg)

Using the latest version:
![image](https://user-images.githubusercontent.com/5526410/158015575-ac64cb6e-4e4b-4946-a5e6-10a4764256dc.jpeg)
